### DISCUSSION TITLE
song select search needs fixes

### DISCUSSION THREAD

**Discussion by cloudiees** [ppy/osu#27494] [Ideas]
Song select is kinda scuffed rn, I'm going to try to list a couple of things I noticed that are inconsistent with stable, or have weird behavior.
- The difficulty slider next to the search bar isnt compatible with mods
- The stars filter isnt compatible with mods
- The != filter operator simply doesn't work < **THE MOST IMPORTANT ONE PLS I BEG FIX THIS**
- The status=r filter is exclusively ranked vs stable where its ranked and approved (tbh not that big of an issue just a weird thing)
- #14499 exists, but I think I should expand on it a little more. The auto switching is really annoying to deal with, as sometimes I just fat finger a key and poopie I lost what map I was on (not really a big deal but still really frustrating). Would be really nice to see the way stable implemented the search and auto switching in lazer
- When you search for a map and two beatmaps of the same set that are adjacent aren't grouped while in difficulty sort 
- Not exactly part of in-game search, but under beatmap listings I feel like an installed vs not installed would be a really good search option

If anyone else has other weird things to point out with the song select search please do!

**Comment by Stoppedpuma** [comment]
> status=r filter is exclusively ranked vs stable where its ranked and approved

IIRC approved is deprecated and will just become ranked at some point

**Reply by cloudiees** [reply]
even if that is the case approved still should be part of status=r until it officially dies imo.

**Comment by bdach** [comment]
> The difficulty slider next to the search bar isnt compatible with mods
> The stars filter isnt compatible with mods

both of these have so many existing reports

> The != filter operator simply doesn't work

is not implemented

> Not exactly part of in-game search, but under beatmap listings I feel like an installed vs not installed would be a really good search option

not technically possible

**Reply by cloudiees** [reply]
The third point you bring up was poorly worded on my part. I meant the in-game beatmap listings (the replacement of osu!direct) not the web beatmap listings. If there really is no way to implement it like it was in stable then that kinda sucks, I see a lot of utility in that option while searching for new beatmaps.
